install/management/appliance.go

```diff
@@ -363,16 +363,42 @@ func (d *Dispatcher) setMacToGuestInfo() error {
 	return nil
 }
 
+func (d *Dispatcher) waitingForMAC(dul time.Duration) (map[string]string, error) {
```

Comment:
Looks like you could use the property collector Wait method, so rather than poll at an interval, you'd be waiting for a push from the property collector.  Have a look at this section of the recently added: VirtualMachine.WaitForNetIP helper: https://github.com/vmware/govmomi/commit/388df2f1d8cb5a0059850fdf10520df7923474ef#diff-aee13781bca461bd01dfe83f8c8fa2ecR257